src/main/java/com/encora/Main.java

```diff
@@ -0,0 +1,12 @@
+package com.encora;
+
+import java.io.IOException;
+
+public class Main {
+
+    public static void main(String[] args) throws IOException {
+        Tree<Wine> tree = new BinarySearchTree<Wine>();
```

Comment:
Interview question: 
From Java 7, generic type is inferred on the instance creation.
Docs: https://docs.oracle.com/javase/7/docs/technotes/guides/language/type-inference-generic-instance-creation.html

So you can replace it to: 
` Tree<Wine> tree = new BinarySearchTree<>();` and will make the same effect.